FlagsmithClient/src/main/java/com/flagsmith/Flagsmith.kt

```diff
@@ -21,27 +32,75 @@ import com.flagsmith.internal.enqueueWithResult
     private val readTimeoutSeconds: Long = 6L,
-    private val writeTimeoutSeconds: Long = 6L
-) {
-    private val retrofit: FlagsmithRetrofitService = FlagsmithRetrofitService.create(
-        baseUrl = baseUrl, environmentKey = environmentKey, context = context, cacheConfig = cacheConfig,
-        requestTimeoutSeconds = requestTimeoutSeconds, readTimeoutSeconds = readTimeoutSeconds, writeTimeoutSeconds = writeTimeoutSeconds)
+    private val writeTimeoutSeconds: Long = 6L,
+    override var lastSeenAt: Double = 0.0 // from FlagsmithEventTimeTracker
+) : FlagsmithEventTimeTracker {
+    private lateinit var retrofit: FlagsmithRetrofitService
+    private var cache: Cache? = null
+    private var lastUsedIdentity: String? = null
     private val analytics: FlagsmithAnalytics? =
         if (!enableAnalytics) null
         else if (context != null) FlagsmithAnalytics(context, retrofit, analyticsFlushPeriod)
         else throw IllegalArgumentException("Flagsmith requires a context to use the analytics feature")
 
+    private val eventService: FlagsmithEventService? =
+        if (!enableRealtimeUpdates) null
+        else FlagsmithEventService(eventSourceUrl = eventSourceUrl, environmentKey = environmentKey) { event ->
+            if (event.isSuccess) {
+                lastEventUpdate = event.getOrNull()?.updatedAt ?: lastEventUpdate
+
+                // Check whether this event is anything new
+                if (lastEventUpdate > lastSeenAt) {
+                    // First evict the cache otherwise we'll be stuck with the old values
+                    cache?.evictAll()
+                    lastSeenAt = lastEventUpdate
+
+                    // Now we can get the new values
+                    getFeatureFlags(lastUsedIdentity) { res ->
+                        if (res.isFailure) {
+                            Log.e(
+                                "Flagsmith",
+                                "Error getting flags in SSE stream: ${res.exceptionOrNull()}"
+                            )
+                        } else {
+                            Log.i("Flagsmith", "Got flags due to SSE event: $event")
+
+                            // If the customer wants to subscribe to updates, emit the new flags
+                            flagUpdateFlow.tryEmit(res.getOrNull() ?: emptyList())
+                        }
+                    }
+                }
+            }
+        }
+
+    // The last time we got an event from the SSE stream or via the API
+    private var lastEventUpdate: Double = 0.0
+
+    /// Stream of flag updates from the SSE stream if enabled
+    val flagUpdateFlow = MutableStateFlow<List<Flag>>(listOf())
+
     init {
         if (cacheConfig.enableCache && context == null) {
             throw IllegalArgumentException("Flagsmith requires a context to use the cache feature")
         }
+        if (enableRealtimeUpdates) {
+            getFlagUpdates()
+        }
+        val pair = FlagsmithRetrofitService.create(
```

Comment:
Would it be possible to tidy this up to something like: 

```kotlin
val (retrofit, cache) = FlagsmithRetrofitService.create(
```

Reply:
Unfortunately you can't do that in Kotlin in a type-safe way - see https://www.baeldung.com/kotlin/returning-multiple-values#no-tuples-in-kotlin.